src/new_index/schema.rs

```diff
@@ -268,18 +284,68 @@ impl Indexer {
         Ok(result)
     }
 
+    fn reorg(&self, reorged: Vec<HeaderEntry>, daemon: &Daemon) -> Result<()> {
+        if reorged.len() > 10 {
+            warn!(
+                "reorg of over 10 blocks ({}) detected! Wonky stuff might happen!",
+                reorged.len()
+            );
+        }
+        // This channel holds a Vec of [u8; 32] scripts found in the blocks (with duplicates)
+        // if we reorg the whole mainnet chain it should come out to about 145 GB of memory.
+        let (tx, rx) = crossbeam_channel::unbounded();
+        // Delete txstore
+        start_fetcher(self.from, daemon, reorged.clone())?
+            .map(|blocks| self.add(&blocks, Operation::DeleteBlocks));
+        // Delete history_db
+        start_fetcher(self.from, daemon, reorged)?
+            .map(|blocks| self.index(&blocks, Operation::DeleteBlocksWithHistory(tx.clone())));
+        // All senders must be dropped for receiver iterator to finish
+        drop(tx);
+
+        // All senders are dropped by now, so the receiver will iterate until the
+        // end of the unbounded queue.
+        let scripts = rx.into_iter().collect::<HashSet<_>>();
+        for script in scripts {
+            // cancel the script cache DB for these scripts. They might get incorrect data mixed in.
+            self.store.cache_db.delete(vec![
```

Comment:
Maybe instead, we should only run delete if we can get() the keys? Not sure which would be faster.

Seems like the extra get() would add extra overhead when we delete, but be faster than attempting a delete if there is no row... averaging out to being faster overall? Would need to measure it...

Probably premature optimization concerns, just noting the concern.

Reply:
I suspect the difference isn't worth worrying about, but we can always change it later if profiling reveals a big difference.